infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -126,6 +127,7 @@ public LogReplicationSinkManager(String localCorfuEndpoint, LogReplicationMetada
                                      LogReplicationContext replicationContext) {
 
         this.replicationContext = replicationContext;
+        String endpoint = CorfuSaasEndpointProvider.getCorfuSaasEndpoint().orElseGet(() -> localCorfuEndpoint);
```

Comment:
we do not need to construct the endpoint again here, right?  This endpoint is coming from the replication context which is set by the SessionManager with the correct endpoint.